src/compiler/parser.ts

```diff
@@ -594,7 +599,7 @@ namespace ts {
             // If this is a javascript file, proactively see if we can get JSDoc comments for
             // relevant nodes in the file.  We'll use these to provide typing informaion if they're
             // available.
-            if (isJavaScript(fileName)) {
+            if (isSourceFileJavaScript(sourceFile)) {
                 addJSDocComments();

```

Comment:
I think the idea was to completely eliminate any major perf hit if you were just using TypeScript.

Additionally, are those the only nodes we'd have to consider? #4447 has a (non-comprehensive) list of places we'd like to have JSDoc template completion.